sdk/core/azure-core-tracing-opentelemetry/src/test/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracerTest.java

```diff
@@ -0,0 +1,304 @@
+        final SpanId parentSpanId = parentSpan.getContext().getSpanId();
+
+        // Act
+        final Context updatedContext = openTelemetryTracer.start(METHOD_NAME, tracingContext, ProcessKind.MESSAGE);
+
+        // Assert
+        // verify span created with explicit parent when no span context in the sending Context object
+        assertSpanWithExplicitParent(updatedContext, parentSpanId);
+        // verify no kind set on Span for message
+        final ReadableSpan recordEventsSpan =
+            (ReadableSpan) updatedContext.getData(PARENT_SPAN_KEY).get();
+        assertEquals(Span.Kind.INTERNAL, recordEventsSpan.toSpanData().getKind());
+        // verify diagnostic id and span context returned
+        assertNotNull(updatedContext.getData(SPAN_CONTEXT_KEY).get());
+        assertNotNull(updatedContext.getData(DIAGNOSTIC_ID_KEY).get());
+    }
+
+    @Test
+    public void startSpanProcessKindProcess() {
+        // Arrange
+        final SpanId parentSpanId = parentSpan.getContext().getSpanId();
+
+        // Act
+        final Context updatedContext = openTelemetryTracer.start(METHOD_NAME, tracingContext, ProcessKind.PROCESS);
+
+        // verify no parent span passed
+        assertFalse(tracingContext.getData(SPAN_CONTEXT_KEY).isPresent(),
+            "When no parent span passed in context information");
+        // verify span created with explicit parent
+        assertSpanWithExplicitParent(updatedContext, parentSpanId);
+        // verify scope returned
+        assertNotNull(updatedContext.getData("scope").get());
+        final ReadableSpan recordEventsSpan =
+            (ReadableSpan) updatedContext.getData(PARENT_SPAN_KEY).get();
+        assertEquals(Span.Kind.SERVER, recordEventsSpan.toSpanData().getKind());
+    }
+
+    @Test
+    public void startProcessSpanWithRemoteParent() {
+        // Arrange
+        final Span testSpan = tracer.spanBuilder("child-span").startSpan();
+        final SpanId testSpanId = testSpan.getContext().getSpanId();
+        final SpanContext spanContext = SpanContext.createFromRemoteParent(
+            testSpan.getContext().getTraceId(),
+            testSpan.getContext().getSpanId(),
+            testSpan.getContext().getTraceFlags(),
+            testSpan.getContext().getTracestate());
+        final Context traceContext = tracingContext.addData(SPAN_CONTEXT_KEY, spanContext);
+
+        // Act
+        final Context updatedContext = openTelemetryTracer.start(METHOD_NAME, traceContext, ProcessKind.PROCESS);
+
+        // Assert
+        assertNotNull(updatedContext.getData("scope").get());
+        // Assert new span created with remote parent context
+        assertSpanWithRemoteParent(updatedContext, testSpanId);
+    }
+
+    @Test
+    public void startSpanOverloadNullPointerException() {
+
+        // Assert
+        assertThrows(NullPointerException.class, () ->
+            openTelemetryTracer.start("", Context.NONE, null));
+    }
+
+    // TODO: Fix links for openTelemetry
```

Comment:
should this test be uncommented at this point?

Reply:
Otel Links are not working as expected, will track this issue here - https://github.com/Azure/azure-sdk-for-java/issues/6188